src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchForm.component.js

```diff
@@ -45,14 +45,45 @@ const Index = ({
     searchViaUniqueIdOnScopeProgram,
     searchViaAttributesOnScopeProgram,
     searchViaAttributesOnScopeTrackedEntityType,
-    selectedOptionId,
+    selectedSearchScopeId,
     classes,
     availableSearchOptions,
     forms,
     searchStatus,
     isSearchViaAttributesValid,
 }: Props) => {
     const [error, setError] = useState(false);
+    const [expandedFormId, setExpandedFormId] = useState(null);
+
+    // each time the user selects new search scope we want the expanded form id to be initialised
+    useEffect(() => {
+        setExpandedFormId(null);
+    },
+    [selectedSearchScopeId],
+    );
+
+    const sortedSearchGroup =
+      selectedSearchScopeId ?
+          availableSearchOptions[selectedSearchScopeId].searchGroups
+              .sort(({ unique: xBoolean }, { unique: yBoolean }) => {
```

Comment:
I would prefer the sorting being done in our preconstructed metadata objects instead. If not we would need to duplicate this code to the relationship search.

Reply:
yeah that is very valid. In the next PR you will see it. I am doing it as part of the refactoring when I basically should have done it here. Apologies the chain was big and sometimes keep track of everything at the same time not so easy

Reply:
Forget what I said I can move it here if you feel this is easier for you to review. Just let me know :)

Reply:
It's fine to switch between PR's, dont worry about it. Will continue the conversion in the next PR.